### PR TITLE
bpo-6532: Make the thread id an unsigned integer.

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
From C API side the type of results of PyThread_start_new_thread() and
PyThread_get_thread_ident(), the id parameter of
PyThreadState_SetAsyncExc(), and the thread_id field of PyThreadState
changed from "long" to "unsigned long".

### CODE REVIEW

File: Modules/_threadmodule.c [hunk 1]
```diff
@@ -1137,13 +1137,8 @@ information about locks.");
...
 thread_get_ident(PyObject *self)
 {
-    long ident;
-    ident = PyThread_get_thread_ident();
-    if (ident == -1) {
-        PyErr_SetString(ThreadError, "no current thread ident");
-        return NULL;
-    }
-    return PyLong_FromLong(ident);
+    unsigned long ident = PyThread_get_thread_ident();
```
Comment (vstinner) [inline Modules/_threadmodule.c]: In _PyImport_ReleaseLock(), you test "== PYTHREAD_INVALID_THREAD_ID". Please be consistent: remove the test in _PyImport_ReleaseLock(), or restore the test it. I'm in favor of dropping the test: pthread_self() and GetCurrentThreadId() cannot fail.

**Review by vstinner — COMMENTED** [review]
LGTM except of a minor issue.

File: Modules/_threadmodule.c (hunk 1 above)
Comment (serhiy-storchaka) [reply Modules/_threadmodule.c]: Good catch. Don't know why I removed the check here. The code was initially written 4 years ago.

**Review by vstinner — APPROVED** [review]

**Comment by vstinner** [comment]
Thanks for your contrib Serhiy! LGTM.

**Pull request merged**